src/pages/ImportData.js

```diff
@@ -119,12 +121,22 @@ const ImportData = () => {
   })
 
   return h(FooterWrapper, [
-    h(TopBar, { title: `Import ${isDataset ? 'Data' : 'Snapshot'}` }),
+    h(TopBar, { title: title || `Import ${isDataset ? 'Data' : 'Snapshot'}` }),
     div({ role: 'main', style: styles.container }, [
       backgroundLogo,
       div({ style: styles.card }, [
-        h2({ style: styles.title }, [`Importing ${isDataset ? 'Data' : `Snapshot ${snapshotName}`}`]),
-        div({ style: { fontSize: 16 } }, ['From: ', new URL(url).hostname]),
+        h2({ style: styles.title }, [header || `Importing ${isDataset ? 'Data' : `Snapshot ${snapshotName}`}`]),
+        supportMultipleImports ?
+          div({ style: { marginTop: 20, marginBottom: 60 } }, [
+            'Dataset(s):',
+            ul({ style: { listStyle: 'none', position: 'relative', marginLeft: 0, paddingLeft: '2rem' } }, [
+              _.map.convert({ cap: false })(({ title, id }, mapindex) => li({ key: `snapshot_${id}`, style: { fontSize: 16, fontWeight: 'bold', marginTop: 20, paddingTop: mapindex ? 20 : 0, borderTop: `${mapindex ? 1 : 0}px solid #AAA` } }, [
+                icon('success-standard', { size: 18, style: { position: 'absolute', left: 0, color: colors.primary() } }),
+                title
+              ]), snapshots)
+            ])
+          ]) :
```

Comment:
We have a util that will get the index for an item when using map: `toIndexPairs`. The only other changes here are formatting:

```suggestion
          div({ style: { marginTop: 20, marginBottom: 60 } }, [
            'Dataset(s):',
            ul({ style: { listStyle: 'none', position: 'relative', marginLeft: 0, paddingLeft: '2rem' } }, [
              _.flow(
                Utils.toIndexPairs,
                _.map(([mapindex, { title, id }]) => li({
                  key: `snapshot_${id}`,
                  style: {
                    fontSize: 16,
                    fontWeight: 'bold',
                    marginTop: 20,
                    paddingTop: mapindex ? 20 : 0,
                    borderTop: `${mapindex ? 1 : 0}px solid #AAA`
                  }
                }, [
                  icon('success-standard', { size: 18, style: { position: 'absolute', left: 0, color: colors.primary() } }),
                  title
                ])))(snapshots)
            ])
          ]) :
```